### DISCUSSION TITLE
SteamClient | Messaging users and friends.

### DISCUSSION THREAD

**Discussion by JBdev05** [ValvePython/steam#449] [General]
Hey guys, I am very confused with how to send messages to my friends with this API. I think I have done everything right, but my friend does not receive the message? In the code below it prints my friends name correctly after getting his SteamUser with getuser(). But the message doesnt get sent??
<img width="626" alt="Screenshot 2023-10-19 105800" src="https://github.com/ValvePython/steam/assets/84608640/e61fad7f-dfe4-4597-aab7-961171260ecb">

**Comment by Gobot1234** [comment]
You need to use client.run_forever()to schedule the send message function on gevents event loop

**Reply by JBdev05** [reply]
ok nice i got it to work with that, but is there any way to grab a friends current game they are playing? PersonaState only returns if they are online or not and stuff.

**Reply by JBdev05** [reply]
or is there any way to send_message in a while true?
I used web api to get the game but now steamclient stuff doesnt work in the while true?
